src/scripts.js

```diff
@@ -53,59 +56,76 @@ const setCustomerData = (customer, rooms, bookings) => {
-  
   hotel.setAvailableRooms(todaysDate);
 
-  const availableRooms = hotel.availableRooms;
-  
+  const availableRooms = hotel.setAvailableRooms(todaysDate);
   domUpdates.displayAvailableRooms(availableRoomsCardsContainer, availableRooms);
 }
 
+const getAvailableRoomsWithInputs = () => {
+  let filteredRoomsByDate;
+  let filteredRoomsByType;
+  let dateInput = customerDateInput.value.split('-').join('/');
+  const todaysDate = hotel.convertTodaysDate();
+  const filterTerm = roomTypesInput.value;
+  
+  if (filterTerm !== '' && dateInput === '') {
+    filteredRoomsByType = hotel.checkAvailableRoomsByType(filterTerm, todaysDate);
+    autofillCurrentDate();
+    displayFilterResults(filteredRoomsByType);
+  } else if (filterTerm === '' && dateInput !== '') {
+    dateInput < todaysDate ? domUpdates.displayApologeticMessage(apologeticMessageContainer) : filteredRoomsByDate = hotel.setAvailableRooms(dateInput);
+    filteredRoomsByDate ? displayFilterResults(filteredRoomsByDate) : domUpdates.clearAvailableRoomsCardsContainer(availableRoomsCardsContainer);
+  } else if (filterTerm !== '' && dateInput !== '') {
+    checkBothInputs(dateInput, todaysDate, filteredRoomsByType, filterTerm);
+  } else {
+    autofillCurrentDate();
+    domUpdates.displayAvailableRoomsView(availableRoomsContainer, pastVisitsContainer, upcomingVisitsContainer, dashboardButton, availableRoomsButton);
+  }
+}
 
+const displayFilterResults = filteredRoomsByInput => {
+  domUpdates.displayAvailableRooms(availableRoomsCardsContainer, filteredRoomsByInput);
+  domUpdates.displayAvailableRoomsView(availableRoomsContainer, pastVisitsContainer, upcomingVisitsContainer, dashboardButton, availableRoomsButton);
+  filteredRoomsByInput.length === 0 ? domUpdates.displayApologeticMessage(apologeticMessageContainer) : domUpdates.resetApologeticMessage(apologeticMessageContainer);
+}
 
+const resetInputs = () => {
+  roomTypesInput.value = '';
+  customerDateInput.value = '';
+}
 
+const autofillCurrentDate = () => {
+  const todaysDate = hotel.convertTodaysDate().split('/').join('-');
+  customerDateInput.value = todaysDate;
+}
 
-
-
-
-
-
-
-
-
-
-
-
-
-
-
-
-
-
-
-
-
-
-
-
-
-
-
+const checkBothInputs = (dateInput, todaysDate, filteredRoomsByType, filterTerm) => {
+  if (filterTerm !== '' && dateInput !== '' && dateInput < todaysDate) {
+    domUpdates.displayApologeticMessage(apologeticMessageContainer);
+    domUpdates.clearAvailableRoomsCardsContainer(availableRoomsCardsContainer);
+  } else if (filterTerm !== '' && dateInput !== '' && dateInput >= todaysDate) {
+    filteredRoomsByType = hotel.checkAvailableRoomsByType(filterTerm, dateInput);
+    displayFilterResults(filteredRoomsByType);
+  }
+}
 
 window.addEventListener('load', getAllData);
 dashboardButton.addEventListener('click', () => {
   domUpdates.displayDashboardView(availableRoomsContainer, pastVisitsContainer, upcomingVisitsContainer, dashboardButton, availableRoomsButton);
```

Comment:
Here's another area that could be cleaned up by moving the query selectors into the domUpdates file